cluster/gce/manifests/etcd.manifest

```diff
@@ -49,14 +49,14 @@
     "command": [
               "/bin/sh",
               "-c",
-              "if [ -e /usr/local/bin/migrate-if-needed.sh ]; then /usr/local/bin/migrate-if-needed.sh 1>>/var/log/etcd{{ suffix }}.log 2>&1; fi; exec /usr/local/bin/etcd --name etcd-{{ hostname }} --listen-peer-urls {{ etcd_protocol }}://{{ hostname }}:{{ server_port }} --initial-advertise-peer-urls {{ etcd_protocol }}://{{ hostname }}:{{ server_port }} --advertise-client-urls http://127.0.0.1:{{ port }} --listen-client-urls http://127.0.0.1:{{ port }} {{ quota_bytes }} --data-dir /var/etcd/data{{ suffix }} --initial-cluster-state {{ cluster_state }} --initial-cluster {{ etcd_cluster }} {{ etcd_creds }} 1>>/var/log/etcd{{ suffix }}.log 2>&1"
+              "if [ -e /usr/local/bin/migrate-if-needed.sh ]; then /usr/local/bin/migrate-if-needed.sh 1>>/var/log/etcd{{ suffix }}.log 2>&1; fi; exec /usr/local/bin/etcd --name etcd-{{ hostname }} --listen-peer-urls {{ etcd_protocol }}://127.0.0.1:{{ server_port }} --initial-advertise-peer-urls {{ etcd_protocol }}://{{ hostname }}:{{ server_port }} --advertise-client-urls http://127.0.0.1:{{ port }} --listen-client-urls http://127.0.0.1:{{ port }} {{ quota_bytes }} --data-dir /var/etcd/data{{ suffix }} --initial-cluster-state {{ cluster_state }} --initial-cluster {{ etcd_cluster }} {{ etcd_creds }} 1>>/var/log/etcd{{ suffix }}.log 2>&1"
```

Comment:
Should advertise-client-urls really be to localhost?

Reply:
Yes - we don't want to expose it anywhere else.

Reply:
Ohh, actually that's not fully true.
It's true for the client URL.

But I think that doing that for peer-urls will actually break e.g. GKE

Reply:
@jpbetz - feel free to reach my offline to discuss this, but that will break things

Reply:
@wojtek-t Yes, I think your (both) right. We should continue to advertise peer-urls via hostname. If we do change `--advertise-client-urls` to localhost, we would also need to update kube-apiserver's flags since it currently had ` --etcd-servers=http://127.0.0.1:2379`. I'll make that a separate change since it's not directly related to this PR and needs to be done carefully if we decide to do it.

Reply:
So I think you need to revert the change this line, right?

Reply:
No, the bind to 127.0.0.1 is still correct. This discussion was only about advertise flags.

Reply:
OK - I think you're right.

Reply:
It stops etcd nodes to communicate when they are running on different machines.
PR with revert - https://github.com/kubernetes/kubernetes/pull/59974